src/logger/MetricsLogger.ts

```diff
@@ -33,6 +33,71 @@ export class MetricsLogger {
     this.context = context || MetricsContext.empty();
   }
 
+  /**
+   * Checks if every value of the given dimension is a string
+   * 
+   * @param {Array} the array of dimensions sets ([{dimensionName: 'dimensionValue'}])
+   * @returns {boolean}
+   */
+  private static checkDimensionValueTypes(dimensionSets : Array<Record<string, string>>) : boolean {
+    return dimensionSets.every((dimensionSet : Record<string, string>) => {
+      const dimensionNames : Array<string> = Object.keys(dimensionSet);
+      return dimensionNames.every((dimensionName : string) => {
+        return typeof dimensionSet[dimensionName] === "string";
+      });
+    });
+  }
+
+  /**
+   * Returns an array of all the existing dimension keys
+   * 
+   * @returns {Array}
+   */
+  private getExistingDimensionNames() : Array<string> {
+    let existingDimensionNames : Array<string> = [];
+    this.context.getDimensions().forEach((existingDimensionSet : Record<string, string>) => {
+      const dimensionNames : Array<string> = Object.keys(existingDimensionSet);
+      existingDimensionNames = existingDimensionNames.concat(dimensionNames); 
+    });
+    return existingDimensionNames;
+  }
+
+  /**
+   * Check if the given dimension already contains a property that is not representing the value of an existing dimension
+   * 
+   * @param {Array} the array of dimensions sets ([{dimensionName: 'dimensionValue'}])
+   * @returns {boolean}
+   */
+  private checkDimensionPropertyConflicts(dimensionSets : Array<Record<string, string>>) : boolean {
+    let dimensionNames : Array<string> = [];
+    dimensionSets.forEach((dimensionSet : Record<string, string>) => {
+      dimensionNames = dimensionNames.concat(Object.keys(dimensionSet));
+    });
```

Comment:
similarly

Reply:
That's a great suggestion! Thank you. It will require the es2019 lib, so I added it in the compiler options (check out the latest commit).

Reply:
AWS CodeBuild is now failing and I'm not sure why. The linter is fine, the tests are passing and the build is successful in my environment when I execute `npm run build`. I don't have access to the details of AWS CodeBuild, but I suspect it has something to do with introducing es2019 support.